File: src/py_pkg/py_pkg/parameters_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class ParametersNode(Node):

    def __init__(self):
        super().__init__("parameters_node")
        self.declare_parameter("message", "Simple publisher")
        self.publisher_ = self.create_publisher(String, "simple_topic", 10)
        self.timer_ = self.create_timer(0.5, self.publish_example)
        self.get_logger().info("Publisher with parameters has been started ...")

    def publish_example(self):
        msg = String()
        msg.data = self.get_parameter("message").value
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = ParametersNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/cpp_pkg/src/parameters_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class ParametersNode : public rclcpp::Node
{
    public:
        ParametersNode() : Node("parameters_node")
        {   
            this->declare_parameter("message", "Simple publisher");
            publisher_ = this->create_publisher<example_interfaces::msg::String>("simple_topic", 10);
            timer_ = this->create_wall_timer(500ms, std::bind(&ParametersNode::publish_example, this));
            RCLCPP_INFO(this->get_logger(), "Publisher with parameters has been started ...");
        }
        
    private:
        void publish_example()
        {
            auto msg = example_interfaces::msg::String();
            msg.data = this->get_parameter("message").as_string();
            publisher_->publish(msg);
        } 
    
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
}; 

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<ParametersNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}